### PR TITLE
Feature/jobbird unit test01

### PR DESCRIPTION
**Pull request by adriaanbakker-java**
changed to junit5, coverage 95 percent

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -87,7 +87,17 @@
...
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (mostafa-221) [inline pom.xml]: JUnit was already in the dependancies and did not need to be added

File: src/main/java/nl/ordina/jobcrawler/scrapers/HTMLStructureException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package nl.ordina.jobcrawler.scrapers;
```
Comment (mostafa-221) [inline src/main/java/nl/ordina/jobcrawler/scrapers/HTMLStructureException.java:1]: Exceptions should be in the controller/exception package

File: src/main/java/nl/ordina/jobcrawler/scrapers/HTMLStructureException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+public class HTMLStructureException extends Exception {
+    public HTMLStructureException(String s) {
+        super("HTML structure altered in a critical way:" + s);
+    }
+}
```
Comment (mostafa-221) [inline src/main/java/nl/ordina/jobcrawler/scrapers/HTMLStructureException.java]: Exceptions should be in the controller/exception package

File: src/main/java/nl/ordina/jobcrawler/service/DocumentService.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+public class DocumentService {
+
+    VacancyScraper vs;
+
+    public Document getDocument(String aSearchURL) {
+        return vs.getDocument(aSearchURL);
+    }
+
+}
```
Comment (mostafa-221) [inline src/main/java/nl/ordina/jobcrawler/service/DocumentService.java:15]: From what I understand, this class forwards the call to the static method in the abstract VacancyScraper. But why not calling the method directly from the VacancyScraper?

File: src/main/java/nl/ordina/jobcrawler/service/LogService.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+public class LogService {
+
+    public void logInfo(String aMessage) {
+        log.info(aMessage);
+    }
+
+    public void logError(String aMessage) {
+        log.error(aMessage);
+    }
+}
```
Comment (mostafa-221) [inline src/main/java/nl/ordina/jobcrawler/service/LogService.java:19]: This is a good idea, to have our own streamlined logging service. However to be consistent we should use it for all logging or not using it. To be discussed in the meeting.

File: src/main/java/nl/ordina/jobcrawler/service/ScraperService.java [hunk 6]
```diff
@@ -12,6 +12,7 @@
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
+import javax.annotation.PostConstruct;
```
Comment (mostafa-221) [inline src/main/java/nl/ordina/jobcrawler/service/ScraperService.java]:
```suggestion
```
as the @PostConstruct is commented. Maybe the annotation should also be removed

**Review by mostafa-221 — COMMENTED** [review]
I am not sure but shouldn't the test resource files of the JobBirdScraperTest be .html instead of .htm?

File: src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java [hunk 7]
```diff
@@ -36,36 +38,45 @@
...
+
+    public void setDocumentService(DocumentService documentService) {
+        this.documentService = documentService;
+    }
 
+    private LogService logService = new LogService();
+    private DocumentService documentService  = new DocumentService();
 
-    private static final int MAX_NR_OF_PAGES = 25;  // 25 seems enough for demo purposes, can be up to approx 60
+    public static final int MAX_NR_OF_PAGES = 25;  // 25 seems enough for demo purposes, can be up to approx 60
```
Comment (mostafa-221) [inline src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java]: I searched the calls of this variable and they are all in the same class, which makes me wonder why this change was done

File: src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java [hunk 8]
```diff
@@ -88,14 +99,26 @@ public JobBirdScraper() {
...
+     * Default function to start scraping vacancies.
+     *
+     * @return List with vacancies.
+     */
+    @Override
+    public List<Vacancy> getVacancies() {
+
+        List<String> vacancyURLs = retrieveURLs();
+        return retrieveVacancies(vacancyURLs);
+    }
```
Comment (mostafa-221) [inline src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java:113]: Nice refactoring

File: src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java [hunk 9]
```diff
@@ -157,7 +181,7 @@ private boolean continueSearching(ArrayList<String> vacancyURLs, ArrayList<Strin
      * @param doc The HTML document containing the URLs to the vacancies
      * @return the index of the last page to scrape
      */
-    private int getLastPageToScrape(Document doc) {
+    protected int getLastPageToScrape(Document doc) throws HTMLStructureException {
         int totalNumberOfPages = getTotalNumberOfPages(doc);
         // TODO: we could get more sophisticated logic in place to limit the number of pages.
         // For example, we could look at the posting date of each vacancy, and limit it to thirty days.
```
Comment (mostafa-221) [inline src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java]: this method does not throw that exception, so I do not think it is necessary

**Comment by mostafa-221** [comment]
The ```getVacancies``` and the throwing of an exception of ```createSearchURL``` were not tested. For the rest it is full, good job!

**Comment by mostafa-221** [comment]
The changing methods to protected for testing is a step that I am not sure about. Maybe something to discuss.

File: src/main/java/nl/ordina/jobcrawler/scrapers/HTMLStructureException.java (hunk 3 above)
Comment (mostafa-221) [reply src/main/java/nl/ordina/jobcrawler/scrapers/HTMLStructureException.java]: Also maybe it is a good idea to have it extend the RuntimeException

File: pom.xml (hunk 1 above)
Comment (adriaanbakker-java) [reply pom.xml]: adapted

File: src/main/java/nl/ordina/jobcrawler/scrapers/HTMLStructureException.java (hunk 3 above)
Comment (adriaanbakker-java) [reply src/main/java/nl/ordina/jobcrawler/scrapers/HTMLStructureException.java]: adapted

File: src/main/java/nl/ordina/jobcrawler/service/DocumentService.java (hunk 4 above)
Comment (adriaanbakker-java) [reply src/main/java/nl/ordina/jobcrawler/service/DocumentService.java:15]: we spoke about this in the scrum meeting and I showed you why this is not possible.

File: src/main/java/nl/ordina/jobcrawler/service/LogService.java (hunk 5 above)
Comment (adriaanbakker-java) [reply src/main/java/nl/ordina/jobcrawler/service/LogService.java:19]: the other code has no direct use for it. It has been introduced here because some tests depend on log mocking

File: src/main/java/nl/ordina/jobcrawler/service/ScraperService.java (hunk 6 above)
Comment (adriaanbakker-java) [reply src/main/java/nl/ordina/jobcrawler/service/ScraperService.java]: adapted

File: src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java (hunk 7 above)
Comment (adriaanbakker-java) [reply src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java]: no specific reason

File: src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java (hunk 8 above)
Comment (adriaanbakker-java) [reply src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java:113]: thx

File: src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java (hunk 9 above)
Comment (adriaanbakker-java) [reply src/main/java/nl/ordina/jobcrawler/scrapers/JobBirdScraper.java]: adapted

**Review by mostafa-221 — APPROVED** [review]
